_posts/2022-12-06-integrating-auth0-with-react.md

```diff
@@ -0,0 +1,307 @@
+---
+layout: post
+font: serif
+title: Integrating Auth0 Authentication into React
+date: 2022-12-06
+highlight: monokai
+author: Chris Howard
+image: img/blog/integrating-auth0-with-react/banner.jpg
+tags: Auth0 react authentication typescript
+description: 
+---
+
+Authentication and Authorization is objectively the most important part of any application, So why is it so often left to the end of development cycles or implemented in a slap dash manner?
+
+It may be due to the difficulty in creating authentication systems and implementing them into our applications, but with the tools and services available to developers it has become easier and easier to create a robust, secure and easy to use system. So let's not wait and get started on integrating [Auth0](https://auth0.com/) into a new React application!
+
+<br />
+
+## Contents
+
+- [Why Auth0?](#why-auth0)
+- [Create Auth0 Application](#create-auth0-application)
+- [Integration into React](#integrate-with-react)
+    * [Login](#but-wait-how-do-users-login)
+    * [Logout](#great-now-how-about-logout)
+- [Conclusion](#conclusion)
+
+## Why Auth0?
+
+Auth0 is one of the most highly reputed company working in this space. Why build your own service when you can get the years of experience and knowledge dedicated to making sure your users login data is safe?
+
+Auth0 provides Authentication, Authorization, Social Logins, MFA, Analytics and lots more, And to make it all the sweeter, has amazing developer documentation. All of this completely free if you have under 7000 users!
+
+## Create Auth0 Application
+
+Let's start with create a new Auth0 Application
+
+Select Applications > Applications from the side menu (or alternatively the Create Application on the Getting Started page)
+
+![Create Application](/img/blog/integrating-auth0-with-react/create_application.png){:width="962px" height="173px" loading="lazy"}
+![Create Application Getting Started](/img/blog/integrating-auth0-with-react/create_application_alt.png){:width="797px" height="277px" loading="lazy"}
+
+Here we will be able to give our Auth0 Application a name, and a type, In our example, we will select **Single Page Web Applications**
+
+![Create Application Popup](/img/blog/integrating-auth0-with-react/create_application_popup.png){:width="600px" height="543px" loading="lazy"}
+
+Once the application has been created, Navigate to the Settings and these are the key credentials we will need to integrate Auth0 into our application
+
+![Application Settings](/img/blog/integrating-auth0-with-react/client_credentials.png){:width="760px" height="330px" loading="lazy"}
+
+One last step before we can begin integrating! We need to let Auth0 know where we are allowed to send users after authentication or logout, These are the **Allowed Callback URLs** and the **Allowed Logout URLs**, in addition, if you would like Auth0 to handle refreshing the users token and session persistence, we will need to configure the **Allowed Web Origins** aswell.
```

Comment:
Style: Exclamation Points

https://docs.google.com/document/d/1W1e0nKku0IYZ5spUXIeI8TG04sAELGvT67CHGHGwJT0/edit#heading=h.1mrcu09

Reply:
```suggestion
One last step before we can begin integrating! We need to let Auth0 know where we are allowed to send users after authentication or logout, These are the **Allowed Callback URLs** and the **Allowed Logout URLs**, in addition, if you would like Auth0 to handle refreshing the users token and session persistence, we will need to configure the **Allowed Web Origins** as well.
```
typo

Reply:
* Allowed Callback URL
* Allowed Logout URL
* Allowed Web Origins

Do you expect the reader to understand what each of the above are?  What values should I put in there?

If you expect the reader to understand, hyperlinks to reference documentation for anyone that is confused would help.  If you don't expect prior knowledge then you probably need to introduce each term